src/httprpc.cpp

```diff
@@ -136,10 +130,14 @@ static bool RPCAuthorized(const std::string& strAuth, std::string& strAuthUserna
     if (!userpass_data) return false;
     strUserPass.assign(userpass_data->begin(), userpass_data->end());
 
-    if (strUserPass.find(':') != std::string::npos)
-        strAuthUsernameOut = strUserPass.substr(0, strUserPass.find(':'));
-
-    return CheckUserAuthorized(strUserPass);
+    size_t colon_pos = strUserPass.find(':');
+    if (colon_pos == std::string::npos) {
+        return false; // Invalid basic auth.
+    }
+    std::string user = strUserPass.substr(0, colon_pos);
+    std::string pass = strUserPass.substr(colon_pos + 1);
```

Comment:
In 98ff38a6f1a8a1e214bd3905a2dcac31ae6c2f52 "rpc: Perform HTTP user:pass split once in `RPCAuthorized`"

Could use `util::Split` instead of reimplementing split

Reply:
i intentionally didn't do that, as that would change the semantics. It currently breaks on the first `:`, which would allow semicolons in the password.